python/lwt_interface/README.md

```diff
@@ -8,4 +8,55 @@ C API in your own compiled Python module (either via Cython
 or the Python C API), you almost certainly don't need to use
 this code.
 
-## TODO document
+## Overview
+
+To allow a tskit table collection to be transferred from one compiled Python
+extension module to another the table collection is converted to a `dict` of
+basic python types and numpy arrays. This is then converted back in the receiving
+module. `tskit_lwt_interface.h` provides a function `register_lwt_class` that 
+defines a Python class `LightweightTableCollection` that performs these conversions
+with methods `asdict` and `fromdict`. These methods mirror the `asdict` and `fromdict`
+methods on `tskit.TableCollection`.
+
+## Usage
+An example C module skeleton `example_c_module.c` is provided, which shows passing tables
+back and forth to the C module. See `test_example_c_module.py` for the python example usage
+of the example module.
+
+To add the 
+`LightweightTableCollection` type to your module you include `tskit_lwt_interface.h`
+and then call `register_lwt_class` on your C Python module object. You can then convert
+to and from the lightweight table collection in Python, for example to convert a tskit
+`TableCollection` to a `LightweightTableCollection`:
+```python
+tables = tskit.TableCollection(1)
+lwt = example_c_module.LightweightTableCollection()
+lwt.fromdict(tables.asdict())
+```
+and vice-versa:
+```python
+tc = tskit.TableCollection(lwt.asdict())
+```    
+In C you can access the tables in a `LightweightTableCollection` instance that is passed 
+to your function:
+```C
+LightweightTableCollection *lwt_tables = NULL;
+PyArg_ParseTuple(args, "O!", &LightweightTableCollectionType, &lwt_tables))
+```
+The variable `lwt_tables->tables` then contains a standard `tsk_table_collection_t`.
+
+To send tables back to python you can create an instance:
+```C    
```

Comment:
This is a bit tricky, as there's a lot of other infrastructure around this you need in order to make it memory safe etc. I think I'd rather keep the example C code chunks out of this file, and we just direct the reader to look at the example module for how to use it.

Reply:
Yep, agreed and removed.